frontend/src/pages/connectionTypes/manage/advanced/FileUploadExtensionRow.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import {
+  Button,
+  FormHelperText,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupItem,
+  TextInput,
+} from '@patternfly/react-core';
+import { ExclamationCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+import { EXTENSION_REGEX } from '~/concepts/connectionTypes/fields/fieldUtils';
+
+type FileUploadExtensionRowProps = {
+  extension?: string;
+  isDuplicate?: boolean;
+  onChange: (newValue: string) => void;
+  onRemove?: () => void;
+  allowRemove: boolean;
+  textRef?: React.RefObject<HTMLInputElement>;
+};
+
+const FileUploadExtensionRow: React.FC<FileUploadExtensionRowProps> = ({
+  extension,
+  isDuplicate,
+  onRemove,
+  allowRemove,
+  onChange,
+  textRef,
+}) => {
+  const [isValid, setIsValid] = React.useState<boolean>(!isDuplicate);
+
+  React.useEffect(
+    () => setIsValid(extension ? !isDuplicate && EXTENSION_REGEX.test(extension) : true),
+    // only run on entry or if a duplicate, otherwise wait for blur
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [isDuplicate],
+  );
```

Comment:
Each row maintains it's own valid state. While on blur the regex validation is performed but duplicate error is done in realtime and passed in. Why not one way or the other?

Reply:
Each row is unaware of the other rows, only the form knows all the rows so it determines if it is a duplicate.